lib/test_track_rails_client/assignment_helper.rb

```diff
@@ -8,14 +8,10 @@ def stub_test_track_assignments(assignment_registry) # rubocop:disable Metrics/A
 
     assignment_registry.each do |split_name, variant|
       prefixed_split_name = "#{app_name}.#{split_name}"
-      if split_registry['splits'].key?(prefixed_split_name)
-        assignments << { split_name: prefixed_split_name, variant: variant.to_s, unsynced: false }
-      else
-        unless split_registry['splits'][split_name]
```

Comment:
The existence of this conditional seems dubious. I can't see why we wouldn't want this split registry override always (regardless of whether the split is already in the registry), because of the visitor-generation scenario I described in the description. 

It also created some unexpected behavior based on whether the split name was passed in as a string or a symbol. Such that if you were trying to stub for one of those visitor-generating contexts and you passed in the split name as a string instead of a symbol, the stub wouldn't actually be applied.

Reply:
This makes sense. I think it probably was just trying to be minimally invasive. For example, we're totally hardcoding the `feature_gate` property to false when we "make up" a split. We should maybe consider calculating that from suffix just to be more legit, if we're going to override the split registry in all cases?

Reply:
yup, i'll make that change.